### PR TITLE
Add FreeBSD CI

### PR DESCRIPTION
**Pull request by slonopotamus**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -35,4 +35,17 @@ jobs:
       with:
         fetch-depth: 1
     - name: "Run unit tests"
-      run: go test -v .
\ No newline at end of file
+      run: go test -v .
+
+  test-unit-freebsd:
+    runs-on: ubuntu-latest
```
Comment (fahedouch) [inline .github/workflows/test.yml:41]: please avoid `latest` version

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -35,4 +35,17 @@ jobs:
       with:
         fetch-depth: 1
     - name: "Run unit tests"
-      run: go test -v .
\ No newline at end of file
+      run: go test -v .
+
+  test-unit-freebsd:
```
Comment (fahedouch) [inline .github/workflows/test.yml:40]: is it possible to combine `test-unit` for multiple os using strategy matrix ?

File: .github/workflows/test.yml (hunk 1 above)
Comment (slonopotamus) [reply .github/workflows/test.yml]: Replaced with 22.04

File: .github/workflows/test.yml (hunk 2 above)
Comment (slonopotamus) [reply .github/workflows/test.yml:40]: I'm not sure how to do that given that I need to use completely different steps. The only common thing is checkout.

Comment (fahedouch) [reply .github/workflows/test.yml:40]: you can be inspired from this https://github.com/containerd/nerdctl/blob/main/.github/workflows/test.yml#L68-L96

Comment (slonopotamus) [reply .github/workflows/test.yml]: I've pushed updated version. To be honest, it looks absolutely disgusting :D

Comment (slonopotamus) [reply .github/workflows/test.yml:43]: Done.

**Review by fahedouch — APPROVED** [review]
Thanks

**Pull request merged**